### PR TITLE
feat: ezETH listing on lido

### PR DESCRIPTION
**Pull request by brotherlymite**
- [ ] Fund the Executor with seed tokens

### CODE REVIEW

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  {
+    IAaveV3ConfigEngine.EModeCategoryUpdate[]
+      memory eModeUpdates = new IAaveV3ConfigEngine.EModeCategoryUpdate[](2);
+
+    eModeUpdates[0] = IAaveV3ConfigEngine.EModeCategoryUpdate({
+      eModeCategory: 2,
+      ltv: 75_00,
+      liqThreshold: 78_00,
+      liqBonus: 7_50,
+      label: 'ezETH USDS correlated'
```
Comment (brotherlymite) [inline src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:85]: Not sure if it should be `LRT Stablecoin correlated` better

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    });
+    eModeUpdates[1] = IAaveV3ConfigEngine.EModeCategoryUpdate({
+      eModeCategory: 3,
+      ltv: 93_00,
+      liqThreshold: 95_00,
+      liqBonus: 1_00,
+      label: 'ezETH wstETH correlated'
```
Comment (brotherlymite) [inline src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:92]: Better name: `LRT LST correlated`?

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      enabledToBorrow: EngineFlags.DISABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
+      withSiloedBorrowing: EngineFlags.DISABLED,
+      flashloanable: EngineFlags.ENABLED,
+      ltv: 5,
+      liqThreshold: 10,
+      liqBonus: 7_50,
+      reserveFactor: 15_00,
+      supplyCap: 15_000,
+      borrowCap: 100,
```
[the remark this answers is not in the document]
Comment (Matt-Graham-TL) [reply src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:57]: Can we add a comment on the forum to catchup this amendment ?

[the remark this answers is not in the document]
Comment (eboadom) [reply src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:57]: cap 0 means no cap @Matt-Graham-TL , it is simply misconfigured to put it that way. I think ACI can comment on the forum when creating the proposal

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+import {AaveV3PayloadEthereumLido} from 'aave-helpers/src/v3-config-engine/AaveV3PayloadEthereumLido.sol';
+import {EngineFlags} from 'aave-v3-origin/contracts/extensions/v3-config-engine/EngineFlags.sol';
+import {IAaveV3ConfigEngine} from 'aave-v3-origin/contracts/extensions/v3-config-engine/IAaveV3ConfigEngine.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {IEmissionManager} from 'aave-v3-origin/contracts/rewards/interfaces/IEmissionManager.sol';
+
+/**
+ * @title Onboard ezETH to Lido Instance
+ * @author BGD Labs (@bgdlabs)
```
Comment (eboadom) [inline src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:14]: ACI will be submitting the proposal, so the author can be changed to their typical authoring format

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.t.sol [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      user
+    );
+  }
+
+  function test_ezETH_wstETH_emode() public {
+    GovV3Helpers.executePayload(vm, address(proposal));
+
+    vm.prank(GovernanceV3Ethereum.EXECUTOR_LVL_1);
+    // as caps have been reached
+    AaveV3EthereumLido.POOL_CONFIGURATOR.setBorrowCap(
```
Comment (eboadom) [inline src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.t.sol:88]: This should not be needed. Test should be of the payload, which increases caps

**Review by eboadom — APPROVED** [review]

File: src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      enabledToBorrow: EngineFlags.DISABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
+      withSiloedBorrowing: EngineFlags.DISABLED,
+      flashloanable: EngineFlags.DISABLED,
+      ltv: 5,
+      liqThreshold: 10,
+      liqBonus: 7_50,
+      reserveFactor: 15_00,
+      supplyCap: 15_000,
+      borrowCap: 100,
```
Comment (sendra) [inline src/20241021_AaveV3EthereumLido_OnboardEzETHToLidoInstance/AaveV3EthereumLido_OnboardEzETHToLidoInstance_20241021.sol:57]: this cap is not mention on the forum. or snapshot

**Review by sendra — APPROVED** [review]

**Pull request merged**